echopop/survey.py

```diff
@@ -213,20 +216,299 @@ def stratified_analysis(
+                    "fit_parameters": fit_parameters,
+                    "model": model,
+                    "standardize_coordinates": standardize_coordinates,
+                    "stratum_name": self.analysis["settings"]["transect"]["stratum_name"],
+                    "variable": variable,
+                    "verbose": verbose,
+                }
+            }
+        )
+
+        # Append `kriging_parameters` to the settings dictionary
+        if standardize_coordinates:
+            self.analysis["settings"]["variogram"].update(
+                {"kriging_parameters": self.input["statistics"]["kriging"]["model_config"]}
+            )
+
+        # Create a copy of the existing variogram settings
+        variogram_parameters = self.input["statistics"]["variogram"]["model_config"].copy()
+
+        # Update the `variogram_parameters` dictionary based on user inputs
+        # ---- Model name
+        variogram_parameters["model"] = model
+        # ---- Nugget
+        if nugget is not None:
+            variogram_parameters["nugget"] = nugget
+        # ---- Sill
+        if sill is not None:
+            variogram_parameters["sill"] = sill
+        # ---- Correlation range
+        if correlation_range is not None:
+            variogram_parameters["correlation_range"] = correlation_range
+        # ---- Hole effect range
+        if hole_effect_range is not None:
+            variogram_parameters["hole_effect_range"] = hole_effect_range
+        # ---- Decay power
+        if decay_power is not None:
+            variogram_parameters["decay_power"] = decay_power
+        # ---- Semivariance enhancement
+        if enhance_semivariance is not None:
+            variogram_parameters["enhance_semivariance"] = enhance_semivariance
+        # ---- Lag resolution
+        if lag_resolution is not None:
+            variogram_parameters["lag_resolution"] = lag_resolution
+        # ---- Max range
+        if max_range is not None:
+            variogram_parameters["range"] = max_range
+        # ---- Number of lags
+        variogram_parameters["n_lags"] = n_lags
+        # ---- Azimuth range
+        variogram_parameters["azimuth_range"] = azimuth_range
+        # ---- Force lag-0
+        variogram_parameters["force_lag_zero"] = force_lag_zero
+
+        # Create optimization settings dictionary
+        # ---- Preallocate if not pre-defined by user
+        if optimization_parameters is None:
+            optimization_parameters = {
+                "max_fun_evaluations": max_fun_evaluations,
+                "cost_fun_tolerance": cost_fun_tolerance,
+                "solution_tolerance": solution_tolerance,
+                "gradient_tolerance": gradient_tolerance,
+                "finite_step_size": finite_step_size,
+                "trust_region_solver": trust_region_solver,
+                "x_scale": x_scale,
+                "jacobian_approx": jacobian_approx,
+            }
```

Comment:
I think we can make the input argument list less daunting by requiring users to put in a dict for updating any of the default params. If I remember correctly it is not recommended to default an input argument to a dict directly (because it is mutable?), but we can use a preset dict for that. Something like:
```python
DEFAULT_OPT_DICT = {
    "param1": value1, ...
}

... # code blocks in the middle

# assumes opt_params below is the user input argument
opt_params_out = DEFAULT_OPT_DICT
if not isinstance(opt_params, dict):  # make sure input is a dict
    raise ValueError("this should be a dict")
if opt_params:  # update using all input params at once
    opt_params_out.update(opt_params)
```